src/backend/catalog/dependency.c

```diff
@@ -641,7 +641,7 @@ recursiveDeletion(const ObjectAddress *object,
 				 * just ereport here, rather than proceeding, since no other
 				 * dependencies are likely to be interesting.)
 				 */
-				if (callingObject == NULL)
+				if (!creating_extension && callingObject == NULL)
```

Comment:
This block of code makes sure that no dependent object is dropped apart from `drop extension`. In gpdb 6 this is handled by `DEPENDENCY_INTERNAL_AUTO` which doesn't exist in gpdb 5, hence we can move it there. `creating_extension` flag is checked to make sure that `drop extension` works correctly, otherwise it will fail.

Reply:
Any that tries to drop dependent object

https://github.com/arenadata/gpdb/blob/4e2f4284c996891977633dc559d8a4d01db236fb/src/test/modules/test_extensions/expected/test_extensions.out#L40-L42

Reply:
I got the same error for the given snippet even without using of `creating_extension` flag.
Do you have a scenario which shows unwanted behavior without this line?

Reply:
Test case was added

Reply:
https://github.com/arenadata/gpdb/blob/599e1286a8caf54431ae39a40bf2f4f3632cb52b/src/test/modules/test_extensions/expected/test_extensions.out#L51-L53

Reply:
The flag was removed, test case was added